fontforgeexe/gotodlg.c

```diff
@@ -91,7 +91,10 @@ static int Goto_OK(GGadget *g, GEvent *e) {
 	gw = GGadgetGetWindow(g);
 	d = GDrawGetUserData(gw);
 	ret = GGadgetGetTitle8(GWidgetGetControl(gw,CID_Name));
-	if ( d->ranges!=NULL ) {
+	d->ret = NameToEncoding(d->sf,d->map,ret);
+	if ( d->ret<0 || (d->ret>=d->map->enccount && d->sf->cidmaster==NULL ))
```

Comment:
Non-rhetorical question: what happens here if `d->ret>=d->map->enccount` but `d->sf->cidmaster!=NULL`?

Reply:
`d->ret>=d->map->enccount` would normally mean that `d->ret` is not a valid encoding slot, but if `d->sf->cidmaster!=NULL` the implication is that it is valid if you take the CID map into account. So, in that case, the function ends up returning the current valid `d->ret`, without popping up the error dialog.

Reply:
Would `d->ret>=d->map->enccount` be an unexpected result when `d->sf->cidmaster==NULL` that the code is protecting against just in case? Otherwise, can there be the analogous problem when `d->sf->cidmaster!=NULL` (it's a CID font but the value is still bogus)?

Reply:
I don’t know. I just swapped the two blocks, so if it was bogus before, it is still bogus.